### PR TITLE
ubuntu-dev: reduce bloat

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
asciidoc drags in the whole texlive suite, which weighs at about 500MB. apt repositories should be also cleaned, while the impact is not large.

### CODE REVIEW

**Comment by ishitatsuyuki** [comment]
Don't merge yet. We seems to have depended on recommended packages.

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 1]
```diff
@@ -23,6 +24,7 @@ RUN __LLVM_VERSION__="6.0" \
  && apt-get update -q \
  && apt-get upgrade -qy \
  && apt-get install -qy \
+  --no-install-recommends \
```
Comment (jankeromnes) [inline ubuntu-dev/ubuntu-dev.dockerfile]: Nit: I'd prefer `apt-get install` options that aren't packages to be on the same line as `apt-get install` (just like `-qy`).

**Comment by ishitatsuyuki** [comment]
Pushed some other changes. If you don't like them I will swap them out.

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 2]
```diff
@@ -59,13 +61,17 @@ RUN __LLVM_VERSION__="6.0" \
   sudo \
   supervisor \
   tmux \
+  unzip \
   valgrind \
+  wget \
   x11vnc \
+  xmlto \
   xvfb \
+ && rm -rf /var/lib/apt/lists/* \
```
[the remark this answers is not in the document]
Comment (notriddle) [reply ubuntu-dev/ubuntu-dev.dockerfile]: Each Dockerfile RUN record creates a layer; in this case, as long as you always run the rm line every time you do an apt-get, the apt lists will never end up in an image layer.

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
 
 # Add source for the latest Neovim packages.
-RUN echo "deb http://ppa.launchpad.net/neovim-ppa/stable/ubuntu xenial main" > /etc/apt/sources.list.d/neovim.list \
- && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 9DBB0BE9366964F134855E2255F96FCF8231B6DD
+RUN add-apt-repository ppa:neovim-ppa/stable
 
 # Install basic development packages.
 RUN __LLVM_VERSION__="6.0" \
- && apt-get update -q \
- && apt-get upgrade -qy \
```
[the remark this answers is not in the document]
Comment (ishitatsuyuki) [reply ubuntu-dev/ubuntu-dev.dockerfile]: Upgrading packages will result in space overhead due to CoW. The ubuntu base image is normally kept almost updated, so there's no reason to hurry.

File: servo/servo.dockerfile [hunk 4]
```diff
@@ -15,11 +14,6 @@ RUN sudo apt-get update -q \
   libglib2.0-dev \
   xorg-dev \
   gperf \
-  g++ \
-  build-essential \
-  cmake \
-  python-virtualenv \
-  python-pip \
```
[the remark this answers is not in the document]
Comment (ishitatsuyuki) [reply servo/servo.dockerfile]: That's a good point. Yes, apt-get install on installed packages is a noop. I will revert it back.

File: firefox/firefox-git.dockerfile [hunk 5]
```diff
@@ -4,10 +4,11 @@ MAINTAINER Jan Keromnes "janx@linux.com"
 # Install Firefox build dependencies.
 # One-line setup command from:
 # https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Build_Instructions/Linux_Prerequisites#Most_Distros_-_One_Line_Bootstrap_Command
-RUN sudo apt-get update -q \
+RUN sudo apt-get update \
```
[the remark this answers is not in the document]
Comment (ishitatsuyuki) [reply firefox/firefox-git.dockerfile]: It's not needed and in the Docker library images it's never used. Often when an error occurs it would be more useful if we had the context (which is suppressed with -q).

File: servo/servo.dockerfile [hunk 6]
```diff
@@ -29,10 +29,12 @@ RUN sudo apt-get update -q \
   libgles2-mesa-dev \
   libegl1-mesa-dev \
   libdbus-1-dev \
-  libpulse-dev \
+  libharfbuzz-dev \
```
Comment (jankeromnes) [inline servo/servo.dockerfile]: Cool drive-by update, thanks!

File: ubuntu-dev/ubuntu-dev.dockerfile [hunk 7]
```diff
@@ -1,28 +1,29 @@
...
  && apt-key add /tmp/llvm-snapshot.gpg.key \
  && rm -f /tmp/llvm-snapshot.gpg.key
 
+# Add source for the latest Git packages.
+RUN add-apt-repository ppa:git-core/ppa
+
 # Add source for the latest Mercurial packages.
-RUN echo "deb http://ppa.launchpad.net/mercurial-ppa/releases/ubuntu xenial main" > /etc/apt/sources.list.d/mercurial.list \
- && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 41BD8711B1F0EC2B0D85B91CF59CE3A8323293EE
+RUN add-apt-repository ppa:mercurial-ppa/releases
```
Comment (jankeromnes) [inline ubuntu-dev/ubuntu-dev.dockerfile]: Note that this makes us a little more vulnerable to malicious key changes (e.g. if the PPA gets hacked).

**Comment by jankeromnes** [comment]
Please also rebase your commits. After that I'll approve this pull request and we can merge it and start saving disk space and CI time!

**Comment by ishitatsuyuki** [comment]
Removed `xmlto` as it's no longer needed with the Git PPA (it was originally installed as recommended package but now we disable them so we needed to list it manually). Ready for review (I hope the CI will be green).

File: dspace/dspace.dockerfile [hunk 8]
```diff
@@ -1,29 +1,45 @@
-FROM janitortechnology/ubuntu-dev
+FROM janx/ubuntu-dev
```
Comment (jankeromnes) [inline dspace/dspace.dockerfile]: Oops, this doesn't look right.
